### PR TITLE
support submit pytorch job

### PR DESCRIPTION
**Pull request by jiaqianjing**
add pytorch-operator api & client about submiting pytorch job #28289726

submit pytorch job #28289726

workerCount - 1, because the master is also considered as a worker #28289726

update charts about submiting pytorch job #28289726

add charts, pytorch operator api&client, and support submit pytorchjob
by git  #28289726

### CODE REVIEW

File: charts/pytorchjob/values.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ttlSecondsAfterFinished:
+
+# rank0
+master: 1
+
+# rankN, is local training when N = 0
+workers: 0
+
+# image addr of worker init container for init-pytorch
+workerInitPytorchImage: alpine:3.10
```
[the remark this answers is not in the document]
Comment (jiaqianjing) [reply charts/pytorchjob/values.yaml:75]: no, we can't now

File: charts/pytorchjob/templates/pytorchjob.yaml [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+    chart: {{ template "pytorchjob.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    createdBy: "PyTorchJob"
+spec:
+  backoffLimit: {{ .Values.retry }}
+{{- if .Values.activeDeadlineSeconds }}
+  activeDeadlineSeconds: {{ .Values.activeDeadlineSeconds }}
+{{- end }}
+{{- if .Values.cleanPodPolicy }}
```
[the remark this answers is not in the document]
Comment (jiaqianjing) [reply charts/pytorchjob/templates/pytorchjob.yaml:20]: this is default of pytorch operator

File: cmd/arena/commands/submit_pytorchjob.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+	//if err != nil {
+	//	log.Debugf("Check %s exist due to error %v", name, err)
+	//}
+	//
+	//if job != nil {
+	//	return fmt.Errorf("the job %s is already exist, please delete it first. use 'arena delete %s'", name, name)
+	//}
+
+	// the master is also considered as a worker
+	submitArgs.WorkerCount = submitArgs.WorkerCount - 1
```
[the remark this answers is not in the document]
Comment (jiaqianjing) [reply cmd/arena/commands/submit_pytorchjob.go:222]: no, user should set workerCount=1

File: cmd/arena/commands/submit_pytorchjob.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+	msg := "the docker image for tensorboard"
+	command.Flags().StringVar(&submitArgs.TensorboardImage, "tensorboardImage", "registry.cn-zhangjiakou.aliyuncs.com/tensorflow-samples/tensorflow:1.12.0-devel", msg)
+	command.Flags().MarkDeprecated("tensorboardImage", "please use --tensorboard-image instead")
+	command.Flags().StringVar(&submitArgs.TensorboardImage, "tensorboard-image", "registry.cn-zhangjiakou.aliyuncs.com/tensorflow-samples/tensorflow:1.12.0-devel", msg)
+
+	command.Flags().StringVar(&submitArgs.TrainingLogdir, "logdir", "/training_logs", "the training logs dir, default is /training_logs")
+
+	// TODO: user can replace "alpine:3.10" with custom image addr
+	command.Flags().StringVar(&submitArgs.WorkerInitPytorchImage,
```
[the remark this answers is not in the document]
Comment (jiaqianjing) [reply cmd/arena/commands/submit_pytorchjob.go:85]: ok

File: cmd/arena/commands/submit_pytorchjob.go [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+		"worker-init-pytorch-image",
+		"alpine:3.10",
+		"the docker image for workers init container image wait for master ready, default 'alpine:3.10'")
+
+	command.Flags().StringVar(&submitArgs.CleanPodPolicy, "clean-task-policy", "None", "How to clean tasks after Training is done, only support None, All.")
```
Comment (cheyang) [inline cmd/arena/commands/submit_pytorchjob.go:90]: Add TODO for running

File: cmd/arena/commands/submit_pytorchjob.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cheyang) [reply cmd/arena/commands/submit_pytorchjob.go:222]: Should it be --worker-count or --count?

File: charts/pytorchjob/templates/pytorchjob.yaml [hunk 6]
```diff
@@ -0,0 +1,566 @@
...
+metadata:
+  name: {{ .Release.Name }}
+  labels:
+    app: {{ template "pytorchjob.name" . }}
+    chart: {{ template "pytorchjob.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    createdBy: "PyTorchJob"
+spec:
+  backoffLimit: {{ .Values.retry }}
```
Comment (cheyang) [inline charts/pytorchjob/templates/pytorchjob.yaml:16]: Remove backoffLimit

File: charts/pytorchjob/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+appVersion: "1.0"
+description: A Helm chart for PyTorchJob
+name: pytorchjob
+version: 0.0.1
```
Comment (happy2048) [inline charts/pytorchjob/Chart.yaml:5]: the version(0.0.1) is not matched the version(0.1.0) of the CHANGELOG.md?

File: charts/pytorchjob/templates/ingress.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
+{{- if .Values.ingress -}}
+apiVersion: extensions/v1beta1
```
Comment (happy2048) [inline charts/pytorchjob/templates/ingress.yaml:2]: versions before 1.14  can use extensions/v1beta1, version after 1.14 use "networking.k8s.io/v1beta1",did you test "extensions/v1beta1" on k8s v1.16.9?

File: charts/pytorchjob/Chart.yaml (hunk 7 above)
Comment (jiaqianjing) [reply charts/pytorchjob/Chart.yaml:5]: ok

File: cmd/arena/commands/submit_pytorchjob.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jiaqianjing) [reply cmd/arena/commands/submit_pytorchjob.go:222]: we can use --worker-count firstly, and waiting for user's fedback

File: cmd/arena/commands/submit_pytorchjob.go (hunk 5 above)
Comment (jiaqianjing) [reply cmd/arena/commands/submit_pytorchjob.go:90]: ok

File: cmd/arena/commands/submit_pytorchjob.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jiaqianjing) [reply cmd/arena/commands/submit_pytorchjob.go:226]: WorkerCount default value：1， so user  needn't to set it when run standalone trainning job

File: charts/pytorchjob/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (cheyang) [reply charts/pytorchjob/values.yaml]: So we need add TODO for this.

**Pull request merged**